Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -70,12 +72,41 @@ extension NavigationMapView {
                                                    selector: #selector(orientationDidChange(_:)),
                                                    name: UIDevice.orientationDidChangeNotification,
                                                    object: nil)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(didUpdateEHorizonRoadName),
+                                                   name: .electronicHorizonDidUpdatePosition,
+                                                   object: nil)
+
         }
         
         private func suspendNotifications() {
             NotificationCenter.default.removeObserver(self,
                                                       name: UIDevice.orientationDidChangeNotification,
                                                       object: nil)
+            NotificationCenter.default.removeObserver(self,
+                                                      name: .electronicHorizonDidUpdatePosition,
+                                                      object: nil)
+        }
+        
+        @objc func didUpdateEHorizonRoadName(_ notification: Notification) {
+            guard let startingEdge = notification.userInfo?[RoadGraph.NotificationUserInfoKey.treeKey] as? RoadGraph.Edge else { return }
```

Comment:
Seems like this method should always set `electronicHorizonRoadNames`, even if only to set it to `nil`, to keep a stale name from lingering if the user transitions from a named road to an unnamed road.

Reply:
The `electronicHorizonRoadNames ` could be empty after the removal of the  `mapboxStreetsRoadClass `. And then the `similarFeature` would be `nil`, it would lead to the `wayNameView` hidden. So it would not cause the stale name kept when transition to an unnamed road.